### ISSUE TITLE
Portrait

### ISSUE THREAD

**Issue by Aftermathh** [oUF-wow/oUF#330]
```
Message: ...\AddOns\AftermathhUI\Libs\oUF\elements\portraits.lua:63: script ran too long
Time: 02/03/17 01:28:41
Count: 3
Stack: ...\AddOns\AftermathhUI\Libs\oUF\elements\portraits.lua:63: in function <...\AddOns\AftermathhUI\Libs\oUF\elements\portraits.lua:43>
(tail call): ?
Interface\AddOns\AftermathhUI\Libs\oUF\ouf.lua:159: in function `func'
Interface\AddOns\AftermathhUI\Libs\oUF\events.lua:48: in function <Interface\AddOns\AftermathhUI\Libs\oUF\events.lua:46>
(tail call): ?
[C]: in function `TargetUnit'
Interface\FrameXML\SecureTemplates.lua:483: in function `handler'
Interface\FrameXML\SecureTemplates.lua:619: in function `SecureActionButton_OnClick'
Interface\FrameXML\SecureTemplates.lua:659: in function <Interface\FrameXML\SecureTemplates.lua:651>

Locals: self = oUF_AftermathhTarget {
 0 = <userdata>
 UNIT_PORTRAIT_UPDATE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\portraits.lua:75
 __tags = <table> {
 }
 PLAYER_ENTERING_WORLD = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\ouf.lua:148
 UNIT_SPELLCAST_CHANNEL_START = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:256
 SpellRange = <table> {
 }
 UNIT_CLASSIFICATION_CHANGED = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\qicon.lua:55
 PhaseIcon = <unnamed> {
 }
 UNIT_MAXHEALTH = <table> {
 }
 __elements = <table> {
 }
 Castbar = <unnamed> {
 }
 UNIT_HEAL_ABSORB_AMOUNT_CHANGED = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\healprediction.lua:151
 UNIT_SPELLCAST_NOT_INTERRUPTIBLE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:204
 UNIT_THREAT_SITUATION_UPDATE = <function> defined @Interface\AddOns\AftermathhUI\Modules\Unitframes\Core.lua:79
 UNIT_HEALTH = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\healprediction.lua:151
 Health = <unnamed> {
 }
 UNIT_COMBAT = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF_Plugins\oUF_CombatFeedback.lua:59
 SmoothBar = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF_Plugins\oUF_Smooth.lua:13
 UNIT_MAXPOWER = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\power.lua:220
 RAID_TARGET_UPDATE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\ricons.lua:58
 UNIT_DISPLAYPOWER = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\power.lua:220
 UNIT_CONNECTION = <table> {
 }
 Debuffs = <unnamed> {
 }
 Portrait = <unnamed> {
 }
 UNIT_HEAL_PREDICTION = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\healprediction.lua:151
 UNIT_POWER_BAR_HIDE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\power.lua:220
 UNIT_POWER_BAR_SHOW = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\power.lua:220
 UNIT_SPELLCAST_START = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:104
 unit = "target"
 UNIT_PHASE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\picon.lua:53
 UNIT_POWER = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\power.lua:220
 UNIT_HEALTH_FREQUENT = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\health.lua:145
 UNIT_ABSORB_AMOUNT_CHANGED = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\healprediction.lua:151
 Power = <unnamed> {
 }
 PvP = <unnamed> {
 }
 UNIT_AURA = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\aura.lua:318
 UNIT_SPELLCAST_CHANNEL_UPDATE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:309
 UNIT_SPELLCAST_DELAYED = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:218
 UNIT_SPELLCAST_INTERRUPTIBLE = <function> defined @Interface\AddOns\AftermathhUI\Libs\oUF\elements\castbar.lua:190
 UNIT_THREAT_LIST_UPDATE = <function> defined @Interface\AddOns\AftermathhUI\Modules\Unitframes\Core.lua:79
 UNIT_SPELLCAST_INTERRUPTED = <functi
```

**Comment by p3lim** [comment]
Please attach the relevant layout code.

**Comment by haste** [comment]
Looking at the call stack I doubt there is much we can do. It errors out on the `portrait:SetUnit()` call on  [line 63](https://github.com/oUF-wow/oUF/blob/7d47fc32f3579a519d141fd0d742b714b14806c2/elements/portraits.lua#L63)

This was previously reported in #295 as well.

**Comment by solor** [comment]
I've seen this portrait error several times, cant reproduce it, totally random.

**Comment by haste** [comment]
I assume it happens due to WoW having to load models from cache and that takes too long time :/

**Comment by p3lim** [comment]
>  TheDanW | p3lim: yeah, looks like the model isn't in memory and something is causing a blocking load during that call, i'll try to get this fixed for 7.2

**Comment by p3lim** [comment]
> TheDanW: we haven't had any luck reproducing the "script ran too long" on SetUnit, if you see any patterns or have specific models maybe let me know and i'll have QA look into it
